pandas/tests/series/indexing/test_datetime.py

```diff
@@ -54,12 +54,14 @@ def test_fancy_setitem():
 
 def test_dti_snap():
     dti = DatetimeIndex(['1/1/2002', '1/2/2002', '1/3/2002', '1/4/2002',
-                         '1/5/2002', '1/6/2002', '1/7/2002'], freq='D')
+                         '1/5/2002', '1/6/2002', '1/7/2002'],
+                        name='my_dti', freq='D')
```

Comment:
Can you parametrize over `name` so that we test both cases where there is and is not a name?  So for `name` you'd basically use: `@pytest.mark.parametrize('name', [None, 'my_dti'])`. And similar for `tz` if you fix that too?

Reply:
Fixed.